src/javascript/app/pages/user/account/settings/account_closure.js

```diff
@@ -0,0 +1,86 @@
+const BinarySocket     = require('../../../../base/socket');
+const localize         = require('../../../../../_common/localize').localize;
+const Url              = require('../../../../../_common/url');
+
+const AccountClosure = (() => {
+    const form_selector = '#form_closure';
+
+    const onLoad = () => {
+        $(form_selector).on('submit', (event) => {
+            event.preventDefault();
+            submitForm();
+        });
+        $('#closure-container').setVisibility(1);
+        $('#other-reason').on('keyup', () => $('#other-reason').removeClass('error-field'));
+    };
+
+    const submitForm = () => {
+        const $btn_submit = $(`${form_selector} #btn_submit`);
+        const reason = getReason();
+        if (reason) {
+            $('#closure_loading').setVisibility(1);
+            $btn_submit.attr('disabled', true);
+
+            const data  = { account_closure: 1, reason };
+
+            BinarySocket.send(data).then((response) => {
+                if (response.error) {
+                    showFormMessage(localize('Sorry, an error occurred while processing your request.'), false);
+                    $btn_submit.attr('disabled', false);
+                } else {
+                    $('#closure_loading').setVisibility(0);
+                    $('#closure-container').setVisibility(0);
+                    $('#msg_main').setVisibility(1);
+
+                    setTimeout(() => window.location.href = Url.urlFor('home'), 10000);
+                }
+            });
+        } else {
+            setTimeout(() => { $btn_submit.removeAttr('disabled'); }, 1000);
+        }
+    };
+
+    const showFormMessage = (localized_msg, is_success) => {
+        if (is_success) {
+            $.scrollTo($('h1#heading'), 500, { offset: -10 });
+            $(form_selector).setVisibility(0);
+            $('#msg_main').setVisibility(1);
+        } else {
+            $.scrollTo($('#reason'), 500, { offset: -20 });
+            $('#msg_form')
+                .attr('class', is_success ? 'success-msg' : 'errorfield')
```

Comment:
`className` should be `error-field` or `errorfield`?

Reply:
It is `errorfield`, cause it's a different class and I guess it was meant for the error message not the textfield

Reply:
then above you should also remove errorfield no error-field

Reply:
I'll remove the conditions since we never call this function on success